icechunk/src/storage/s3.rs

```diff
@@ -117,8 +159,20 @@ pub async fn mk_client(config: &S3Options, credentials: S3Credentials) -> Client
         }
     }
 
-    let s3_builder =
+    let mut s3_builder =
         Builder::from(&aws_config.load().await).force_path_style(config.force_path_style);
+
+    if !extra_read_headers.is_empty() || !extra_write_headers.is_empty() {
+        s3_builder = s3_builder.interceptor(ExtraHeadersInterceptor {
+            extra_read_headers,
+            extra_write_headers,
+        })
+    }
+
+    if config.allow_http {
+        s3_builder = s3_builder.force_path_style(true);
+    }
```

Comment:
need to merge `main`

Reply:
oh I thought i did ... did I fuck up the merge? will check

Reply:
fixed